src/Compilers/Core/Portable/SourceGeneration/Nodes/SyntaxValueProvider_ForAttributeWithSimpleName.cs

```diff
@@ -20,6 +21,23 @@ namespace Microsoft.CodeAnalysis;
 
 public partial struct SyntaxValueProvider
 {
+    /// <summary>
+    /// Information computed about a particular tree.  Cached so we don't repeated recompute this important information
+    /// each time the incremental pipeline is rerun.
+    /// </summary>
+    private record SyntaxTreeInfo(SyntaxTree Tree, bool ContainsGlobalAliases, bool ContainsAttributeList);
+
+    /// <summary>
+    /// Caching of syntax-tree to the info we've computed about it.  Used because compilations will have thousands of
+    /// trees, and the incremental pipeline will get called back for *all* of them each time a compilation changes.  We
+    /// do not want to continually recompute this data over and over again each time that happens given that normally
+    /// only one tree will be different.  We also do not want to create an IncrementalValuesProvider that yield this
+    /// information as that will mean we have a node in the tree that scales with the number of *all syntax trees*, not
+    /// the number of *relevant syntax trees*. This can lead to huge memory churn keeping track of a high number of
+    /// trees, most of which are not going to be relevant.
+    /// </summary>
+    private static readonly ConditionalWeakTable<SyntaxTree, SyntaxTreeInfo> s_treeToInfo = new();
```

Comment:
note: as an alternative to a CWT, i could also use a normal dictionary created inside ForAttributeWithSimpleName and then captured.  But i wasn't certain what the semantics were for the lifetimes of these pipelines.  @chsienki what are the lifetimes?  And are the pipelines executed concurrently?